test/src/e2e_vm_tests/test_programs/should_fail/type_mismatch_error_message/test.toml

```diff
@@ -1,3 +1,15 @@
 category = "fail"
 
-# check: $()No method named "does_not_exist" found for type "Result<Data<bool>, str[4]>".
+# check: $()error
+# check: should_fail/type_mismatch_error_message/src/main.sw:18:5
```

Comment:
```suggestion
# check: type_mismatch_error_message/src/main.sw:18:5
```
We're planning on removing the `should_pass` and `should_fail` dirs one day I think.  IMO it's etter not to encode them in the tests.